server/tests/backend_tests.py

```diff
@@ -1,115 +1,112 @@
 
-    def upload_game(self, data_file):
-        with open(data_file) as f:
-            game_str = f.read()
 
-        response = self.client.post('/submit_game', data=game_str, content_type='application/json')
-        assert response.status_code == 201
-
-    def get_stats(self):
-        response = self.client.get('/api/1/stats')
-        stats = response.json
-        return stats
-
-    # Tests
+class StatsSnapshotTests(TestBase, FlaskTest, SnapShotTest):
     def test_basic_point(self):
-        self.upload_game('data/test/basic_point.json')
+        self.upload_game('basic_point.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
     def test_callahan(self):
-        self.upload_game('data/test/callahan.json')
+        self.upload_game('callahan.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
     def test_catch_d(self):
-        self.upload_game('data/test/catch_d.json')
+        self.upload_game('catch_d.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
     def test_drop(self):
-        self.upload_game('data/test/drop.json')
+        self.upload_game('drop.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
     def test_half(self):
-        self.upload_game('data/test/half.json')
+        self.upload_game('half.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
     def test_mini_game(self):
-        self.upload_game('data/test/mini_game.json')
+        self.upload_game('mini_game.json')
 
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
+
     def test_mini_game2(self):
-        self.upload_game('data/test/mini_game2.json')
+        self.upload_game('mini_game2.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
     def test_throw_away(self):
-        self.upload_game('data/test/throw_away.json')
+        self.upload_game('throw_away.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
     def test_turnovers(self):
-        self.upload_game('data/test/turnovers.json')
+        self.upload_game('turnovers.json')
         stats = self.get_stats()
         self.assertMatchSnapshot(stats)
 
 
-    def test_league_endpoint(self):
+class PerfTests(TestBase, FlaskTest):
+    @app.after_request
+    def after_request(response):
+        global queries
+        queries = get_debug_queries()
+
+        global query_count
+        query_count = len(queries)
+
+        return response
+
+
+    # this endpoint doesn't make any queries
+    # this shows that test setup etc is creeping in here
+    def test_current_league_queries(self):
+        self.client.get('/current_league')
+        self.assertEqual(query_count, 19)
+
+
+    def test_game_upload_queries(self):
+        self.upload_game('mini_game.json')
+        self.assertEqual(query_count, 201)
```

Comment:
comparing the numbers here it does look like `upload_game` makes a ton of queries - which makes sense it is inserting every player and their stats. I do wonder if this could be batched properly / better.